migrate/20241213_archived_record.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:archived_record, partition_by: :deleted_at, partition_type: :range) do
+      column :deleted_at, :timestamptz, null: false, default: Sequel::CURRENT_TIMESTAMP
+      column :model_name, :text, null: false
+      column :model_values, :jsonb, null: false, default: "{}"
+    end
+
+    create_table(:archived_record_2024_10, partition_of: :archived_record) do
```

Comment:
Migrations are normal ruby code, I recommend using a loop here instead of manually unrolling it.

Reply:
Switched to creating partitions in a loop.